compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala

```diff
@@ -238,25 +239,29 @@ final class Conversions(
                 builder.setCrash(s"Contract Id comparability Error")
               case NonComparableValues =>
                 builder.setComparableValueError(proto.Empty.newBuilder)
-              case Limit(limitError) =>
-                limitError match {
-                  case Limit.ValueNesting(_) =>
-                    builder.setValueExceedsMaxNesting(proto.Empty.newBuilder)
-                  // TODO https://github.com/digital-asset/daml/issues/11691
-                  //   Handle the other cases properly.
-                  case _ =>
-                    builder.setCrash(s"A limit was overpass when building the transaction")
+              case Dev(_, devError) if devMode =>
+                devError match {
+                  case Dev.Limit(limitError) =>
+                    limitError match {
+                      case Dev.Limit.ValueNesting(_) =>
+                        builder.setValueExceedsMaxNesting(proto.Empty.newBuilder)
+                      // TODO https://github.com/digital-asset/daml/issues/11691
+                      //   Handle the other cases properly.
+                      case _ =>
+                        builder.setCrash(s"A limit was overpassed when building the transaction")
+                    }
+                  case Dev.ChoiceGuardFailed(coid, templateId, choiceName, byInterface) =>
```

Comment:
I realise you are preserving existing behaviour.

But going forward... is there any need to add cases for new Dev errors?

Reply:
We want to add new Dev error like the `WronglyTypeSoftFetch` from @moisesackerman-da's #16839